controllers/openshift/openshift.go

```diff
@@ -1,347 +0,0 @@
-			o.Spec.Template.Spec.InitContainers[0].Command = []string{}
-		}
-	case *corev1.Secret:
-		if allowedNamespace(cr.ObjectMeta.Namespace, os.Getenv("ARGOCD_CLUSTER_CONFIG_NAMESPACES")) {
-			logv.Info("configuring cluster secret with empty namespaces to allow cluster resources")
-			delete(o.Data, "namespaces")
-		}
-	case *rbacv1.Role:
-		if o.ObjectMeta.Name == cr.Name+"-"+"argocd-application-controller" {
-			logv.Info("configuring policy rule for Application Controller")
-
-			// can move this to somewhere common eventually, maybe init()
-			k8sClient, err := initK8sClient()
-			if err != nil {
-				logv.Error(err, "failed to initialize kube client")
-				return err
-			}
-
-			clusterRole, err := k8sClient.RbacV1().ClusterRoles().Get(context.TODO(), "admin", metav1.GetOptions{})
-			if err != nil {
-				logv.Error(err, "failed to retrieve Cluster Role admin")
-				return err
-			}
-			policyRules := getPolicyRuleForApplicationController()
-			policyRules = append(policyRules, clusterRole.Rules...)
-			o.Rules = policyRules
-		}
-	}
-	return nil
-}
-
-func getPolicyRuleForApplicationController() []rbacv1.PolicyRule {
-	return []rbacv1.PolicyRule{
-		{
-			APIGroups: []string{
-				"*",
-			},
-			Resources: []string{
-				"*",
-			},
-			Verbs: []string{
-				"get",
-				"list",
-				"watch",
-			},
-		},
-		{
-			APIGroups: []string{
-				"monitoring.coreos.com",
-			},
-			Resources: []string{
-				"*",
-			},
-			Verbs: []string{
-				"*",
-			},
-		},
-	}
-}
-
-// For OpenShift, we use a custom build of Redis provided by Red Hat
-// which requires additional args in comparison to stock redis.
-func getArgsForRedhatRedis() []string {
-	return []string{
-		"redis-server",
-		"--protected-mode",
-		"no",
-	}
-}
-
-// For OpenShift, we use a custom build of haproxy provided by Red Hat
-// which requires a command as opposed to args in stock haproxy.
-func getCommandForRedhatRedisHaProxy() []string {
-	return []string{
-		"haproxy",
-		"-f",
-		"/usr/local/etc/haproxy/haproxy.cfg",
-	}
-}
-
-// For OpenShift, we use a custom build of Redis provided by Red Hat
-// which requires additional args in comparison to stock redis.
-func getArgsForRedhatHaRedisServer() []string {
-	return []string{
-		"redis-server",
-		"/data/conf/redis.conf",
-	}
-}
-
-// For OpenShift, we use a custom build of Redis provided by Red Hat
-// which requires additional args in comparison to stock redis.
-func getArgsForRedhatHaRedisSentinel() []string {
-	return []string{
-		"redis-sentinel",
-		"/data/conf/sentinel.conf",
-	}
-}
-
-// For OpenShift, we use a custom build of Redis provided by Red Hat
-// which requires additional args in comparison to stock redis.
-func getArgsForRedhatHaRedisInitContainer() []string {
-	return []string{
-		"sh",
-		"/readonly-config/init.sh",
-	}
-}
-
-// policyRulesForClusterConfig defines rules for cluster config.
-func policyRulesForClusterConfig() []rbacv1.PolicyRule {
```

Comment:
So with removing this, would the Operator only ever be able to reconcile a namespace-scoped instance? If the users would want cluster scope, would they need to put in the roles by themselves?

Reply:
This code is actually irrelevant to the argocd-operator itself at the moment since we don't include the openshift package when building the argocd-operator. argocd-operator behavior would remain unchanged

The hook (when registered) overwrites the app-controller clusterrole privileges to something that is acceptable to openshift
But the default clusterrole itself [currently has permissions to do everything on all resources](https://github.com/argoproj-labs/argocd-operator/blob/569db9415782c3c27cb13ddb01e9e3fdb2c31e1e/controllers/argocd/policyrule.go#L9) (maybe something we should look into also?)